src/screens/Analytics/PaymentsAnalytics/PaymentAnalytics.res

```diff
@@ -144,17 +147,98 @@ let make = () => {
     ->JSON.stringify
   }
 
+  let setInitialFilters = HSwitchRemoteFilter.useSetInitialFilters(
+    ~updateExistingKeys,
+    ~startTimeFilterKey,
+    ~endTimeFilterKey,
+    ~origin="analytics",
+    (),
+  )
+
+  React.useEffect0(() => {
+    setInitialFilters()
+    None
+  })
+
+  let startTimeVal = filterValueJson->getString("startTime", "")
+  let endTimeVal = filterValueJson->getString("endTime", "")
+
+  let filterUri = `${Window.env.apiBaseUrl}/analytics/v1/filters/${domain}`
+
+  let filterBody = React.useMemo3(() => {
+    let filterBodyEntity: AnalyticsUtils.filterBodyEntity = {
+      startTime: startTimeVal,
+      endTime: endTimeVal,
+      groupByNames: tabKeys,
+      source: "BATCH",
+    }
+    AnalyticsUtils.filterBody(filterBodyEntity)
+  }, (startTimeVal, endTimeVal, tabKeys->Array.joinWith(",")))
+
+  React.useEffect3(() => {
+    setFilterDataJson(_ => None)
+    if startTimeVal->LogicUtils.isNonEmptyString && endTimeVal->LogicUtils.isNonEmptyString {
+      try {
+        updateDetails(filterUri, filterBody->JSON.Encode.object, Post, ())
```

Comment:
`filterBody->JSON.Encode.object`
Can you define this outside the useEffect as it is being used in the dependencies as well